.devcontainer/Dockerfile

```diff
@@ -71,9 +71,10 @@ ARG PRECOMMIT_VERSION="2.21.0"
 RUN python3 -m pip install pre-commit=="${PRECOMMIT_VERSION}" --no-cache-dir
 
 # Install Python dependencies for data transform pipeline
+ARG SPARK_VERSION="3.3.1"
 ARG NUMPY_VERSION="1.24.2"
 ARG DATABRICKS_CLI_VERSION="0.17.4"
-RUN python3 -m pip install build numpy=="${NUMPY_VERSION}" databricks-cli=="${DATABRICKS_CLI_VERSION}" --no-cache-dir
+RUN python3 -m pip install pyspark==${SPARK_VERSION} pyodbc build numpy=="${NUMPY_VERSION}" databricks-cli=="${DATABRICKS_CLI_VERSION}" --no-cache-dir
```

Comment:
Looks like we've got an ARM problem with `pyodbc`:

```
#0 74.17   Building wheel for pyodbc (pyproject.toml): started
#0 74.31   Building wheel for pyodbc (pyproject.toml): finished with status 'error'
#0 74.32   error: subprocess-exited-with-error
#0 74.32   
#0 74.32   × Building wheel for pyodbc (pyproject.toml) did not run successfully.
#0 74.32   │ exit code: 1
#0 74.32   ╰─> [14 lines of output]
#0 74.32       running bdist_wheel
#0 74.32       running build
#0 74.32       running build_ext
#0 74.32       building 'pyodbc' extension
#0 74.32       creating build
#0 74.32       creating build/temp.linux-aarch64-cpython-310
[2023-02-21T10:47:45.010Z] 
#0 74.32       creating build/temp.linux-aarch64-cpython-310/src
#0 74.32       gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -fPIC -DPYODBC_VERSION=4.0.35 -I/usr/local/include/python3.10 -c src/buffer.cpp -o build/temp.linux-aarch64-cpython-310/src/buffer.o -Wno-write-strings
#0 74.32       In file included from src/buffer.cpp:12:
#0 74.32       src/pyodbc.h:56:10: fatal error: sql.h: No such file or directory
#0 74.32          56 | #include <sql.h>
#0 74.32             |          ^~~~~~~
#0 74.32       compilation terminated.
#0 74.32       error: command '/usr/bin/gcc' failed with exit code 1
#0 74.32       [end of output]
#0 74.32   
#0 74.32   note: This error originates from a subprocess, and is likely not a problem with pip.
#0 74.32   ERROR: Failed building wheel for pyodbc
#0 74.32 Successfully built pyspark databricks-cli
#0 74.32 Failed to build pyodbc
#0 74.32 ERROR: Could not build wheels for pyodbc, which is required to install pyproject.toml-based 
[2023-02-21T10:47:45.012Z] projects
#0 74.83
```

Reply:
I'll have a go at trying to fix

Reply:
As discussed in Teams, requires the `unixodbc-dev` package to be apt-installed before the pip steps

Reply:
hopefully fixed

Reply:
Tested and fixed :)